dace/frontend/python/parser.py

```diff
@@ -129,8 +129,30 @@ def parse_from_function(function, *compilation_args, strict=None):
         # control flow detection in code generation
         xfh.split_interstate_edges(sdfg)
 
-    # Save the SDFG (again)
-    sdfg.save(os.path.join('_dacegraphs', 'program.sdfg'))
+    sdfg_out_path = os.path.join('_dacegraphs', 'program.sdfg')
+
+    if function.cli_args is not None and len(function.cli_args) > 0:
+        os.makedirs(sdfg.build_folder, exist_ok=True)
+        with open(os.path.join(sdfg.build_folder,
+                  'program.sdfgl'), 'w') as launchfiles_file:
+            launchfiles_file.write(
+                'name,SDFG_intermediate,SDFG,source,' +
+                ','.join(
+                    ['argv_' + str(i) for i in range(len(function.cli_args))]
+                ) + '\n'
+            )
+            launchfiles_file.write(
+                sdfg.name + ',' +
+                os.path.abspath(os.path.join(sdfg.build_folder, 'program.sdfg'))
+                + ',' + os.path.abspath(sdfg_out_path) + ',' +
+                os.path.abspath(function.cli_args[0]) + ',' +
+                ','.join([str(el) for el in function.cli_args])
+            )
+
+    # Save the SDFG (again). Skip this step if running from a cached SDFG, as
+    # it might overwrite the cached SDFG.
+    if Config.get_bool('compiler', 'use_cache'):
```

Comment:
this if condition seems inverted

Reply:
Good catch, thanks! Simplification gone wrong..